### ISSUE TITLE
SSH Keys are not deleted during machine removal

### ISSUE THREAD

**Issue by nnatter** [JonasProgrammer/docker-machine-driver-hetzner#55]
Hey,
we recently moved our GitLab runner to Hetzner and are really happy with the result. Thanks a lot for implementing this driver!
Yesterday we noticed a big list of SSH Keys in the Hetzner cloud console. While this should hopefully not cause any problems, it would be nice to automatically clean the obsolete SSH keys from time to time.

I saw that the README contains the following sentence:

> Also note that the driver will attempt to delete the linked key during machine removal, unless `--hetzner-existing-key-id` was used during creation.

Unfortunately, it looks like this is not the case for us. Is there any additional configuration we need to add?

We are using version `3.1.0` of the driver with the following options:
```
      "hetzner-api-token=TOKEN",
      "hetzner-image=ubuntu-20.04",
      "hetzner-server-type=cx21",
      "hetzner-server-location=fsn1",
      # docker-machine is not compatible with docker 20.10 yet
      # https://gitlab.com/gitlab-org/ci-cd/docker-machine/-/issues/41
      "engine-install-url=https://releases.rancher.com/install-docker/19.03.9.sh"
```

Thanks a lot in advance!

**Comment by JonasProgrammer** [comment]
Hi,

thanks for the report, I indeed screwed up when changing the dangling key removal logic for 3.1.0. Funnily enough, it should work for additional keys, but I forgot to test the most basic use case -- sory for the inconvenience.

Anyways, should be fixed in the upcoming 3.1.1 release. Feel free to re-open, if the issue still arises.

Cheers

**Comment by nnatter** [comment]
Hey, just installed version `3.1.1` and the deletion of SSH keys works as expected now. Thanks a lot for fixing this so fast - we are really appreciating your work!
